app/controllers/nodes_controller.rb

```diff
@@ -18,12 +18,31 @@ def show
     @minion = Minion.find(params[:id])
   end
 
+  def assign_roles
+    Minion.assign_roles(roles: { master: [params[:hostname]] })
```

Comment:
In this case there's no security issue (we are explicitly accessing :hostname, but it's a good practice).